docs/source/cookbooks/reproducer.md

```diff
@@ -1,3 +1,159 @@
+to change it.
+
+##### cifmw_job_uri
+This is the magic parameter: by setting it, you'll instruct the reproducer to switch into
+"reproducer job mode". This URI is the one you can find on a Zuul job output, and usually
+looks like this:
+
+https://logserver.rdoproject.org/74/574/LONG_HASH/github-check/podified-multinode-edpm-e2e-nobuild-tagged-crc/SHORT_HASH/
+
+Please take extra care to ensure it's NOT pointing to any sub-directory of that page, such
+as "/controller" - else, it will fail to fetch the needed pieces.
+
+#### Generated files
+There will be a fair amount of generated files, as you may expect. As usual, most of them
+are nested in the `~/ci-framework-data` directory.
+
+On your local host (ansible controller), you will find `~/ci-framework-data/ci-reproducer`
+containing a series of directories. Those directories are generated using the last
+hash in the job URL. It will contain the fetched files from zuul, namely its inventory,
+and the generated parameters nested in the job `ci-framework/artifacts/parameters`
+directory.
+
+On the hypervisor, you'll find the same `~/ci-framework` base directory. Inside it,
+the virtual machine base image will be under the "images" directory, while the layers
+will be in the "workload" one. The reproducer data, such as generated inventory and
+network data, will be there as well. Those will be pushed onto the controller-0
+instance.
+
+#### Access to the virtual machines
+Your local ssh configuration file will be modified by the reproducer in order to offer
+an easy access to the nodes: `ssh controller-0`, `ssh core@crc-0`, `ssh compute-0` and
+so on will work right out of the box.
+
+### Specific parameters
+* `cifmw_reproducer_run_job`: (Bool) Run the actual CI job. Defaults to `true`. If set
+to `false` you'll end with a ready-to-use environment, just before the actual job starts.
+* `cifmw_reproducer_params`: (Dict) Specific parameters you want to pass to the reproducer.
+Defaults to `{}`.
+
+You can refer to the [README](../roles/reproducer.md) nested in the role for more information.
+
+### Running the deploy manually
+The reproducer will generate a playbook mimicking the CI job itself. You can find it under
+`~/src/github.com/openstack-k8s-operators/ci-framework/JOB_ID_play.yml`, where JOB_ID is
+the last hash of the `cifmw_job_uri` parameter. You should, as well, get the proper
+inventory ready in the `scenarios/centos-9/zuul_inventory.yml` file, meaning you should
+be able to run:
+```Bash
+$ ansible-playbook \
+    -i scenarios/centos-9/zuul_inventory.yml \
+    JOB_ID_play.yml
+```
+You will be able to follow the live state in `~/ansible.log`:
+```Bash
+$ tail -f ~/ansible.log
+```
+
+Note that `tmux` is installed by default, and may be handy in such a case ;).
+
+### Cleaning the deployed job
+Just run the following to clean things up:
+```Bash
+$ ansible-playbook \
+    -i custom/inventory.yml \
+    -e cifmw_target_host=builder1 \
+    reproducer-clean.yml
```

Comment:
Where is that file?

Reply:
in the repository... don't tell me I missed its addition??

Reply:
(it was there before : https://github.com/openstack-k8s-operators/ci-framework/blob/main/reproducer-clean.yml - no update to it)

Reply:
Ouch, true, I wasn't expecting that, sorry